cmd/soroban-cli/src/config/sign_with.rs

```diff
@@ -0,0 +1,61 @@
+use crate::{signer, xdr::TransactionEnvelope};
+use clap::arg;
+
+use super::{
+    locator,
+    network::{self, Network},
+    secret,
+};
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error(transparent)]
+    Network(#[from] network::Error),
+    #[error(transparent)]
+    Signer(#[from] signer::Error),
+    #[error(transparent)]
+    Secret(#[from] secret::Error),
+    #[error(transparent)]
+    Locator(#[from] locator::Error),
+    #[error(transparent)]
+    Rpc(#[from] soroban_rpc::Error),
+    #[error("No sign with key provided")]
+    NoSignWithKey,
+    #[error(transparent)]
+    StrKey(#[from] stellar_strkey::DecodeError),
+}
+
+#[derive(Debug, clap::Args, Clone, Default)]
+#[group(skip)]
+pub struct Args {
+    /// Sign with a local key. Can be an identity (--sign-with-key alice), a secret key (--sign-with-key SC36…), or a seed phrase (--sign-with-key "kite urban…"). If using seed phrase, `--hd-path` defaults to the `0` path.
+    #[arg(long, conflicts_with = "sign_with_lab", env = "STELLAR_SIGN_WITH_KEY")]
+    pub sign_with_key: Option<String>,
```

Comment:
this is quite verbose. Should we introduce a short flag like `-s`?

Reply:
We have prematurely added short flags in the past and who is to say that `-s` could mean sign with ledger?

Reply:
We can always add short flags later if it turns out folks really need it. Lets start with self describing commands and see how they are adopted.

Alternatively we could reduce it to `--key`.

Originally when we were adding the `--sign-with-*` we were thinking about how to introduce the sign options on other commands. But now it seems we are only doing custom signing on this command so `--sign-with-*` does seem repetitive given the command is called `tx sign`.

Reply:
I'm removing the suggestion to go with `--key`. Going with `--sign-with-key` gives us the option to add that same optino to other commands in the future, so I think we should stick with it. We can add aliases and shorten it later if we decide this was the wrong call.